can/interfaces/gs_usb.py

```diff
@@ -0,0 +1,107 @@
+from typing import cast, Any, Iterator, List, Optional, Sequence, Tuple, Union
+
+from gs_usb.gs_usb import GsUsb
+from gs_usb.gs_usb_frame import GsUsbFrame
+from gs_usb.constants import CAN_ERR_FLAG, CAN_RTR_FLAG, CAN_EFF_FLAG, CAN_MAX_DLC
+import can
+import usb
+
+
+class GsUsbBus(can.BusABC):
+    def __init__(self, channel, bus, address, bitrate, can_filters=None, **kwargs):
+        """
+        :param channel: usb device name
+        :param bus: number of the bus that the device is connected to
+        :param address: address of the device on the bus it is connected to
+        :param can_filters: not supported
+        :param bitrate: CAN network bandwidth (bits/s)
+        """
+        gs_usb = GsUsb.find(bus=bus, address=address)
+        if not gs_usb:
+            raise can.CanError("Can not find device {}".format(channel))
+        self.gs_usb = gs_usb
+        self.channel_info = channel
+
+        self.gs_usb.set_bitrate(bitrate)
+        self.gs_usb.start()
+
+        super().__init__(channel=channel, can_filters=can_filters, **kwargs)
+
+    def send(self, msg: can.Message, timeout: Optional[float] = None):
+        """Transmit a message to the CAN bus.
+
+        :param Message msg: A message object.
+        :param timeout: timeout is not used here
+
+        :raises can.CanError:
+            if the message could not be sent
+        """
+        can_id = msg.arbitration_id
+
+        if msg.is_extended_id:
+            can_id = can_id | CAN_EFF_FLAG
+
+        if msg.is_remote_frame:
+            can_id = can_id | CAN_RTR_FLAG
+
+        if msg.is_error_frame:
+            can_id = can_id | CAN_ERR_FLAG
+
+        frame = GsUsbFrame()
+        frame.can_id = can_id
+        frame.can_dlc = msg.dlc
+        frame.timestamp_us = int(msg.timestamp * 1000000)
+        msg.data.extend([0x00] * (CAN_MAX_DLC - len(msg.data)))
+        frame.data = list(msg.data)
```

Comment:
I'd finish the data processing and then construct your frame:

```suggestion
        # Pad message data
        msg.data.extend([0x00] * (CAN_MAX_DLC - len(msg.data)))
        
        frame = GsUsbFrame()
        frame.can_id = can_id
        frame.can_dlc = msg.dlc
        frame.timestamp_us = int(msg.timestamp * 1000000)
        frame.data = list(msg.data)
```

You are modifying the message which the user may still be using (e.g. storing/displaying etc) I'd like to avoid that, or at the very least clearly mention in the interface docs.